source/exe/signal_action.h

```diff
@@ -47,7 +49,9 @@ namespace Envoy {
 class SignalAction : NonCopyable {
 public:
   SignalAction()
-      : guard_size_(sysconf(_SC_PAGE_SIZE)), altstack_size_(guard_size_ * 4), altstack_(nullptr) {
+      : guard_size_(sysconf(_SC_PAGE_SIZE)),
+        altstack_size_(std::max(guard_size_ * 4, static_cast<size_t>(MINSIGSTKSZ))),
```

Comment:
I think it's correct. We ultimately mmap `altstack_size_ + 2 * guard_size_` bytes (see `mapSizeWithGuards`) so `altstack_size_` doesn't include the guard pages. The second mprotect calls protects `guard_size_` bytes at an address that's `guard_size_ + altstack_size_` bytes from the start of the mapped memory.

We call `sigaltstack` with `altstack_ + guard_size_` (to skip the first guard page) and `altstack_size_` (which doesn't count the guard pages, and so stops just before the second guard page).

Did I miss something?